File: C++/Solution41.cpp
#include<iostream>
#include<vector>

#include<algorithm>
using namespace std;

class Solution41 {
public:
    int firstMissingPositive(vector<int>& nums) {
        for(int i = 0; i < nums.size(); i++){
            while(nums[i] > 0 && nums[i] <= nums.size() && nums[i] != nums[nums[i] - 1]){
                int temp = nums[nums[i] - 1];
                nums[nums[i] - 1] = nums[i];
                nums[i] = temp;
            }
        }
        for(int i = 0; i < nums.size(); i++){
            if(nums[i] != i + 1){
                return i + 1;
            }
        }
        return nums.size() + 1;
    }
};

int main(){
    vector<int> test = {1, 1};
    Solution41 s;
    cout << s.firstMissingPositive(test);
    return 0;
}

File: Java/other/Solution41.java
class Solution41 {
    public int firstMissingPositive(int[] nums) {
    	boolean[] showup = new boolean[nums.length + 1];
    	for(int ele: nums){
    		if(ele >= 1 && ele <= nums.length){
    			showup[ele] = true;
    		}
    	}
    	for(int i = 1; i < showup.length; i++){
    		if(!showup[i]){
    			return i;
    		}
    	}
    	return showup.length;
    }
}